src/hooks/useNetworkMotionStatesAllColonies.ts

```diff
@@ -0,0 +1,173 @@
+
+export type MotionId = {
+  motionId: string;
+  colonyAddress: string;
+  databaseMotionId: string;
+};
+
+const getVotingReputationAddressByColony = (colony) => {
+  const colonyExtensions = colony?.extensions?.items?.filter(notNull);
+  if (!colonyExtensions) {
+    return [];
+  }
+
+  const currentExtensionAddress = colonyExtensions.find((extension) => {
+    const extensionConfig = supportedExtensionsConfig.find(
+      (e) => getExtensionHash(e?.extensionId) === extension?.hash,
+    );
+    return extensionConfig?.extensionId === Extension.VotingReputation;
+  });
+  return currentExtensionAddress?.address;
+};
+
+const useJoinedColoniesWithExtensions = (userAddress?: string) => {
+  const { data, loading } = useGetJoinedColoniesExtensionsQuery({
+    variables: {
+      contributorAddress: userAddress ?? ADDRESS_ZERO,
+    },
+    skip: !userAddress,
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const joinedColoniesWithExtensions = useMemo(() => {
+    return (
+      data?.getContributorsByAddress?.items
+        .filter(notNull)
+        .map((contributor) => contributor.colony) ?? []
+    );
+  }, [data?.getContributorsByAddress?.items]);
+
+  return {
+    joinedColoniesWithExtensions,
+    loading,
+  };
+};
+
+/**
+ * Hook that accepts an array of motion IDs object and returns a map of motion IDs to their states
+ * Make sure to memoize the array of motion IDs to avoid infinite loops
+ */
+const useNetworkMotionStatesAllColonies = (
+  motionIdsMap: MotionId[],
+  skip?: boolean,
+) => {
+  const { wallet } = useAppContext();
+
+  const {
+    joinedColoniesWithExtensions,
+    loading: joinedColoniesWithExtensionsLoading,
+  } = useJoinedColoniesWithExtensions(wallet?.address);
+
+  const votingReputationByColony = useMemo(() => {
+    return joinedColoniesWithExtensions.reduce((prev, colony) => {
+      if (!colony) {
+        return prev;
+      }
+      return {
+        ...prev,
+        [colony.colonyAddress]: getVotingReputationAddressByColony(colony),
+      };
+    }, {});
+  }, [joinedColoniesWithExtensions]);
+
+  const [loading, setLoading] = useState(false);
+
+  const [motionStatesMap, setMotionStatesMap] = useState<MotionStatesMap>(
+    new Map(),
+  );
+  useEffect(() => {
+    const { ethersProvider } = wallet || {};
+    const signer = ethersProvider?.getSigner(wallet?.address); // Properly initialize the signer with the current wallet address
+
+    const votingReputationAddressesCount = Object.values(
+      votingReputationByColony,
+    ).filter(notUndefined).length;
+    if (
+      skip ||
+      !motionIdsMap.length ||
+      !votingReputationAddressesCount ||
+      !ethersProvider ||
+      !signer
+    ) {
+      return;
+    }
+
+    const newMotionIds = motionIdsMap.filter((motion) => {
+      return !motionStatesMap.has(motion.databaseMotionId);
+    });
+    const deletedMotionIds = Array.from(motionStatesMap.keys()).filter(
+      (nativeMotionKey) =>
```

Comment:
Please update the naming to indicate if it is the native ID or database ID. It should be database ID as only those are unique between colonies